convert2rhel/pkgmanager/handlers/yum.py

```diff
@@ -0,0 +1,247 @@
+__metaclass__ = type
+
+import logging
+import os
+import re
+
+from convert2rhel import pkgmanager
+from convert2rhel.backup import remove_pkgs
+from convert2rhel.pkghandler import get_system_packages_for_replacement
+from convert2rhel.pkgmanager.handlers.base import TransactionHandlerBase
+from convert2rhel.systeminfo import system_info
+from convert2rhel.utils import BACKUP_DIR
+
+
+loggerinst = logging.getLogger(__name__)
+
+# Limit the number of loops overithe yum transaction.
+MAX_NUM_OF_ATTEMPTS_TO_RESOLVE_DEPS = 3
+
+# Extract the first package that appears in the yum depsolve error
+EXTRACT_PKG_FROM_YUM_DEPSOLVE = re.compile(r".*?(?=requires)")
+
+
+def _resolve_yum_problematic_dependencies(output):
+    """Internal function to parse yum resolve dependencies errors.
+
+    This internal function has the purpose to parse the yum `resolveDeps()`
+    errors that may arise from the transaction. It should not be used by other
+    functions here, as the purpose of this is serving the yum single
+    transactions.
+
+    :param output: A list of strings with packages names that had a dependency
+    error.
+    :type output: list[str]
+    """
+    packages_to_remove = []
+    for package in output:
+        resolve_error = re.findall(EXTRACT_PKG_FROM_YUM_DEPSOLVE, str(package))
+        if resolve_error:
+            # The first string to appear in index 0 is the package we want.
+            packages_to_remove.append(str(resolve_error[0]).replace(" ", ""))
+
+    if packages_to_remove:
+        packages_to_remove = set(packages_to_remove)
+        loggerinst.debug(
+            "Removing problematic packages to continue with the conversion:\n%s\n", ", ".join(packages_to_remove)
+        )
+        remove_pkgs(
+            pkgs_to_remove=set(packages_to_remove),
+            backup=True,
+            critical=False,
+            set_releasever=False,
+            reposdir=BACKUP_DIR,
+            manual_releasever=system_info.version.major,
+            varsdir=os.path.join(BACKUP_DIR, "yum/vars"),
+        )
+
+        loggerinst.info("Finished backing up and removing the packages.")
+    else:
+        loggerinst.info("No packages to remove.")
+
+
+class YumTransactionHandler(TransactionHandlerBase):
+    """Implementation of the YUM transaction handler.
+
+    This class will implement and override the public methods that comes from
+    the abstractclass `TransactionHandlerBase`, whith the intention to provide
+    the defaults necessary for handling and processing the yum transactions in
+    the best way possible.
+
+    _base: yum.YumBase()
+        The actual instance of the `yum.YumBase()` class.
+    _enabled_repos: list[str]
+        A list of packages that are used during the transaction.
+    """
+
+    def __init__(self):
+        self._base = None
+
+    def _setup_base(self):
```

Comment:
```suggestion
    def _set_up_base(self):
```
Setup is a noun. To set something up (an activity), one should use "set up".

Reply:
done.